iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractComponentImpl.java

```diff
@@ -44,11 +46,15 @@ public int getSize() {
 
     @Override
     public Iterable<Bus> getBuses() {
-        return Iterables.filter(networkRef.get().getBusView().getBuses(), bus -> bus.getConnectedComponent() == ComponentImpl.this);
+        return StreamSupport.stream(networkRef.get().getBusView().getBuses().spliterator(), false)
+                            .filter(filterOnThis())
+                            .collect(Collectors.toList());
     }
 
     @Override
     public Stream<Bus> getBusStream() {
-        return networkRef.get().getBusView().getBusStream().filter(bus -> bus.getConnectedComponent() == ComponentImpl.this);
+        return networkRef.get().getBusView().getBusStream().filter(filterOnThis());
     }
+
+    protected abstract Predicate<Bus> filterOnThis();
```

Comment:
We should rename this method to give it a better name like getBusFilter() or getBusPredicate()

Reply:
I did it, if it is okay for you, you can merge.